addons/mail/static/src/discuss/core/common/thread_model_patch.js

```diff
@@ -257,7 +257,7 @@ const threadPatch = {
         }
         if (correspondents.length === 0 && this.channel_member_ids.length === 1) {
             // Self-chat.
-            return this.channel_member_ids[0];
+            return this.selfMember;
```

Comment:
I'm not fan of this change and the condition on which it belongs.

correspondent is js knowledge so it might be partial.

Returning self member when we know only one member is not necessarily correct.